### PR TITLE
feat: switch to poetry and add cd pipeline

### PR DESCRIPTION
**Pull request by MelvinKl**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,6 +21,7 @@ When developing, you can use a feature of `pip` called _Editable Installs_,
 which installs the local files like a package so you can test features without
 reinstalling the package every time.
 This package also defines additional depencies for development, like testing and linting.
+It requires `poetry` to be installed.
```
Comment (joaopalet) [inline README.md]: Maybe make it a link to poetry install guide?

Comment (joaopalet) [reply README.md]: This? https://python-poetry.org/docs/#installation

File: .github/workflows/cd.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    runs-on: "ubuntu-latest"
+    steps:      
+      - name: Checkout
+        uses: actions/checkout@v4     
+      - name: Install Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.8"
+      - name: Build & Publish to PyPi
+        uses: JRubics/poetry-publish@v2.0
```
Comment (joaopalet) [inline .github/workflows/cd.yaml:22]: This is alternative to using the `poetry` commands directly, right?

Comment (MelvinKl) [reply .github/workflows/cd.yaml:22]: Yes that's correct.

**Review by joaopalet — APPROVED** [review]

**Pull request merged**